src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -179,9 +179,14 @@ String getCharset() {
         protected FileMonitoringController(FilePath ws) throws IOException, InterruptedException {
             // can't keep ws reference because Controller is expected to be serializable
             ws.mkdirs();
-            FilePath cd = tempDir(ws).child("durable-" + Util.getDigestOf(UUID.randomUUID().toString()).substring(0,8));
-            cd.mkdirs();
-            controlDir = cd.getRemote();
+            FilePath tmpDir = WorkspaceList.tempDir(ws);
+            if (tmpDir != null) {
+                FilePath cd = cd.child("durable-" + Util.getDigestOf(UUID.randomUUID().toString()).substring(0,8));
+                cd.mkdirs();
+                controlDir = cd.getRemote();
+            } else {
+                controlDir = null;
```

Comment:
null seems ok because there is code that handles it https://github.com/jenkinsci/durable-task-plugin/blob/7b093d48504b8a922e40739847cfec8fc455b7fc/src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java#L356-L369

Reply:
Does it make sense to fall back to those code paths though? IIUC, they are for use by old versions of the plugin, so it feels strange to start using them again. What was the old behavior in this case?

Reply:
> What was the old behavior in this case?

Dot-file subdirectories, as shown here.

Reply:
https://github.com/jenkinsci/jenkins/pull/4526#issuecomment-591437539 noted the possibility of using a location in the agent home directory as a fallback for a temp dir when the workspace was a FS root. While that might work out for freestyle projects, it would not work well with Pipeline steps providing a `LauncherDecorator`, like `withDockerContainer` or (K8s) `container`. OTOH those steps are Linux-only and the most plausible case of a root workspace is a drive letter on Windows; `/` would rarely be writable on Linux. (Even on Windows, I would not recommend using `ws` / `dir` to run stuff in a drive root. Rather include `cd` in a batch/Powershell script, or refer to absolute paths.)

Reply:
Oh and my objection to using a subdirectory for `$WORKSPACE_TMP`—that it is a security risk—does not apply to the files managed by `FileMonitoringTask`.